src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,951 @@
+            CancellationToken cancellationToken)
+        {
+            var isParameterized = await ShouldExpressionDisplayCodeActionAsync(document, expression, cancellationToken).ConfigureAwait(false);
+            if (!isParameterized)
+            {
+                return null;
+            }
+            else
+            {
+                var actionsBuilder = ImmutableArray.CreateBuilder<CodeAction>();
+                var actionsBuilderAllOccurrences = ImmutableArray.CreateBuilder<CodeAction>();
+                var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+                actionsBuilder.Add(CreateNewCodeAction(FeaturesResources.and_update_call_sites_directly, allOccurrences: false, trampoline: false, overload: false));
+                actionsBuilderAllOccurrences.Add(CreateNewCodeAction(FeaturesResources.and_update_call_sites_directly, allOccurrences: true, trampoline: false, overload: false));
+
+                if (methodSymbol.MethodKind is not MethodKind.Constructor)
+                {
+                    actionsBuilder.Add(CreateNewCodeAction(
+                        FeaturesResources.into_extracted_method_to_invoke_at_call_sites, allOccurrences: false, trampoline: true, overload: false));
+                    actionsBuilderAllOccurrences.Add(CreateNewCodeAction(
+                        FeaturesResources.into_extracted_method_to_invoke_at_call_sites, allOccurrences: true, trampoline: true, overload: false));
+
+                    actionsBuilder.Add(CreateNewCodeAction(
+                        CreateDisplayText(expression, syntaxFacts, FeaturesResources.into_new_overload_of_0), allOccurrences: false, trampoline: false, overload: true));
+                    actionsBuilderAllOccurrences.Add(CreateNewCodeAction(
+                        CreateDisplayText(expression, syntaxFacts, FeaturesResources.into_new_overload_of_0), allOccurrences: true, trampoline: false, overload: true));
+                }
+
+                return (actionsBuilder.ToImmutable(), actionsBuilderAllOccurrences.ToImmutable());
+            }
+
+            // Local function to create a code action with more ease
+            MyCodeAction CreateNewCodeAction(string actionName, bool allOccurrences, bool trampoline, bool overload)
+            {
+                return new MyCodeAction(actionName, c => IntroduceParameterAsync(
+                    document, expression, methodSymbol, containingMethod, allOccurrences, trampoline, overload, c));
+            }
+        }
+
+        /// <summary>
+        /// Finds the matches of the expression within the same block.
+        /// </summary>
+        protected static async Task<IEnumerable<TExpressionSyntax>> FindMatchesAsync(Document originalDocument,
+            TExpressionSyntax expressionInOriginal, SyntaxNode withinNodeInCurrent,
+            bool allOccurrences, CancellationToken cancellationToken)
+        {
```

Comment:
if `allOccurrences` is false you can just return `expressionInOriginal` right?

Reply:
right